pkg/certagent/agent.go

```diff
@@ -108,17 +111,31 @@ func GenerateCSRObject(config CSRConfig) (*capi.CertificateSigningRequest, error
 
 // RequestCertificate will create a certificate signing request for a node
 // with the config given and send it to a signer via a POST request.
-// If something goes wrong it returns an error.
+// If something goes wrong it returns an error unless CSRConfig.MaxRetry
+// is defined in which case we will retry the request.
 // NOTE: This method does not return the approved CSR from the signer.
 func (c *CertAgent) RequestCertificate() error {
 	csr, err := GenerateCSRObject(c.config)
 	if err != nil {
 		return fmt.Errorf("error generating CSR Object: %v", err)
 	}
 
-	// send CSR to the signer
-	if _, err := c.client.Create(csr); err != nil {
-		return fmt.Errorf("error sending CSR to signer: %v", err)
+	backoff := wait.Backoff{
+		Steps:    c.config.MaxRetry,
+		Duration: 10 * time.Second,
+		Factor:   0.0,
```

Comment:
- whats the behavior at 0?
- why not non-zero value?

Reply:
Well my hope was to be able to know how much time the wait operation would take. By using 0.0 we are not multiplying so if we have 3 steps or `--max-retry=3` this is pretty simple. 

10 seconds initial wait + ( 3 * Steps)  = 40 seconds.

I understand that I am not using the `Exponential` part of this, maybe it was silly to try to keep so simple?

Reply:
https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Backoff